frame/fast-unstake/src/tests.rs

```diff
@@ -108,9 +146,17 @@ fn deregister_works() {
 	});
 }
 
+#[test]
+fn deregister_eras_0_fails() {
```

Comment:
```suggestion
fn deregister_fails_if_era_to_check_not_set() {
```

and the correct(er) way to test this is to register someone successfully, and see that they cannot even be deregistered if the pallet is in halt.

Reply:
well, maybe we can allow deregister at all times and only block register if the pallet is halted?